power_grid_model_c/power_grid_model_c/src/model.cpp

```diff
@@ -79,6 +79,11 @@ void PGM_calculate(PGM_Handle* handle, PGM_PowerGridModel* model, PGM_Options co
         auto const calculation_method = static_cast<CalculationMethod>(opt->calculation_method);
         switch (opt->calculation_type) {
         case PGM_power_flow:
+            if (opt->tap_changing_strategy != PGM_tap_changing_strategy_disabled &&
+                opt->experimental_features == PGM_experimental_features_disabled) {
+                // this option is experimental and should not be exposed to the user
+                throw MissingCaseForEnumError{"PGM_TapChangingStrategy", opt->tap_changing_strategy};
+            }
```

Comment:
one thought i have is that we can check even earlier in the process and instead of raising a `MissingCaseForEnum`, to raise an `ExperimentalFeature` if it is an experimental combination.

The danger of this, is that we first allow it (because it is ignored), then add it as a disallowed experimental feature and then allow it again. The 2nd step would be a breaking change.

To fix that, the `get_optimizer` function should get the `calculation_type` as input. Then, we can mark it as experimental in the C API and it will raise a `MissingCaseForEnum` in the `get_optimizer` function for state estimation/short circuit calculation

Alternatively, we also allow automatic tap optimization with state estimation. While the logic would allow this, it is not a correct approach, because it would be a variable in the math solver itself.